src/models/reposyncstate.ts

```diff
@@ -98,9 +181,23 @@ export default class RepoSyncState extends Sequelize.Model {
 		});
 	}
 
-	setFromRepoJson(repo?: RepositoryData): RepoSyncState {
+	static async updateRepoForSubscription(subscription: Subscription, repo?: RepositoryData): Promise<RepoSyncState | undefined> {
+		const repoId = Number(repo?.repository?.id);
+		if (!repoId) {
+			return undefined;
+		}
+		const model: RepoSyncState | undefined = await RepoSyncState.findOne({
+			where: {
+				subscriptionId: subscription.id,
+				repoId
+			}
+		});
+		return model?.setFromRepositoryData(repo)?.save();
+	}
+
+	setFromRepositoryData(repo?: RepositoryData): RepoSyncState {
 		if (repo) {
-			this.repoUpdatedAt = new Date(repo.repository?.updated_at || Date.now());
+			this.repoUpdatedAt = new Date(repo.repository?.updated_at ?? Date.now());
```

Comment:
Why use `??` instead of `||` here?

Reply:
`??` is null coalescing since we're looking for if it's `undefined` or `null`.  A test that we use is using `new Date(0)` because we wanted to compare the specific 8601 date and instead of using a random number, 0 seemed easy, but 0 is falsy, hence it would return `Date.now()` and fail the test.